### PR TITLE
Fix iniagent regexp chrooting

### PR DESCRIPTION
**Pull request by jreidinger**
https://bugzilla.suse.com/show_bug.cgi?id=962566
see commit message for https://github.com/yast/yast-core/commit/5560a1bb196b0d98f8550663904c92732d9ee11e

### CODE REVIEW

**Comment by teclator** [comment]
LGTM!

**Comment by mvidner** [comment]
Does this need to be fixed for SLE12 too?

**Comment by jreidinger** [comment]
@mvidner I explained in bug report that not as
1) scr switching is done only in installer
2) we do not get any report, so looks like no visible problems found, but I would like to fix it, to prevent potential future problems that can hit us hard.

**Comment by mvidner** [comment]
Yes, I see, now that I've read the bug report. OK.

**Comment by mvidner** [comment]
It would be best if all occurrences of "file", "filename", "fn" in IniParser.h (and less importantly .cc) made it easy to see whether that path includes the SCR chroot or not.

**Comment by jreidinger** [comment]
@mvidner any idea how to make it easily visible? some prefix? like `target_`?

**Comment by mvidner** [comment]
In `SCRAgent` we are using `char * root()` and `string targetPath(string)`, which I think are not really helpful names to distinguish the two kinds of paths.
I think the one `targetPath` returns should include "target" in the name. The other can remain unmarked, or say "source" or "logical". I think "local" has become too associated with "in the inst-sys" so we should not use it.

**Comment by jreidinger** [comment]
@mvidner I have problem to easily recognize then, so for this pr are you ok if I change only start_parser to use `target_fn` instead of `fn`?

**Comment by jreidinger** [comment]
@mvidner all FIXMEs resolved

**Comment by mvidner** [comment]
I've found another area that needs consideration: should we use logical or target filenames *in logging and error messages*?

File: agent-ini/src/IniParser.cc [hunk 1]
```diff
@@ -534,15 +534,15 @@ int IniParser::parse()
...
-			y2error ("Cannot open %s.", *f);
+		    if (scanner_start (*tgt_f))
+			y2error ("Cannot open %s.", *tgt_f);
 		    else
 		    {
-			y2debug ("File %s changed. Reloading.", *f);
-			FileDescr fdsc (agent.targetPath(*f).c_str());
-			multi_files [*f] = fdsc;
+			y2debug ("File %s changed. Reloading.", *tgt_f);
+			FileDescr fdsc (agent.targetPath(*tgt_f).c_str());
```
Comment (mvidner) [inline agent-ini/src/IniParser.cc]: Now that the names are clear, it turns out we are calling `targetPath` once too many here. Our diligence has paid off!

File: agent-ini/src/IniParser.cc [hunk 2]
```diff
@@ -1001,18 +1001,18 @@ int IniParser::write()
 }
 
 // return 0 on success, like write
-int IniParser::write_file(const string & filename, IniSection & section)
+int IniParser::write_file(const string & logical_filename, IniSection & section)
 {
     // ensure that the directories exist
-    Pathname pn(filename);
+    Pathname pn(logical_filename);
```
Comment (mvidner) [inline agent-ini/src/IniParser.cc]: Hmm, it looks like the `targetPath` below should be moved up here.

**Comment by jreidinger** [comment]
@mvidner updated

**Comment by mvidner** [comment]
OK, but some error messages report logical filenames and some report target filenames. We should make that consistent, ideally making it explicit by making the root stand out:
> "It's wrong! (/mnt)/etc/grub3",
> "It's wrong! (/)/etc/grub3"

**Comment by jreidinger** [comment]
@mvidner well, I think it is related, but different issue, not regression, so I think we should not too much extend this pr, especially when I hope that we replace ini agent by cfa with augeas

**Comment by mvidner** [comment]
Since I had an overview of the code, fixing the messages took me little time.
Now it LGTM :)

**Comment by jreidinger** [comment]
thanks, so lets merge it

**Pull request merged**